.github/workflows/publish-charm.yaml

```diff
@@ -0,0 +1,47 @@
+name: Publish charm
+
+on:
+  workflow_call:
+    secrets:
+      CHARMCRAFT_AUTH:
+        required: true
+
+jobs:
+  publish-charm:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Install charmcraft
+        run: sudo snap install charmcraft --classic
+
+      - name: Fetch Tested Charm
```

Comment:
Does this mean you are building the charm on PR and also on merge?

Reply:
We run the whole CI again on merge, and we make sure the charm is tested again before being published.
We had some issues here in the past. When multiple PRs are merged at the same time.

Reply:
Apart from the fact that less compute resources can by used if the charm is built only once, building twice means there are no guarantees that the resulting artifacts are exactly the same. You can check how we are doing it [here](https://github.com/canonical/operator-workflows)

Reply:
I looked into the repo, it probably solves our issues as it looks for a "tested" artifact.
Exploring this repo and using the workflow files from there is in a ticket in our backlog now.

Reply:
If you think it doesn't suit your needs we'd really appreciate the feedback. Thank you!

Reply:
Sure, I'll make sure to let you know once we look into it :)